core/store/ledgerstore/tx_handler.go

```diff
@@ -91,38 +88,27 @@ func (self *StateStore) HandleInvokeTransaction(store store.LedgerStore, overlay
 		BlockHash: block.Hash(),
 	}
 
-	var (
-		costGasLimit      uint64
-		availableGasLimit uint64
-		err               error
-	)
-
-	availableGasLimit = tx.GasLimit
-
 	//init smart contract info
 	sc := smartcontract.SmartContract{
 		Config:       config,
 		CacheDB:      cache,
 		Store:        store,
 		GasTable:     gasTable,
-		Gas:          availableGasLimit,
+		Gas:          tx.GasLimit,
 		WasmExecStep: sysconfig.DEFAULT_WASM_MAX_STEPCOUNT,
 		PreExec:      false,
 	}
 
 	//start the smart contract executive function
 	engine, _ := sc.NewExecuteEngine(invoke.Code, tx.TxType)
 
-	_, err = engine.Invoke()
+	_, err := engine.Invoke()
 	if err != nil {
 		return err
 	}
-	costGasLimit = availableGasLimit - sc.Gas
 
-	var notifies []*event.NotifyEventInfo
 	notify.Notify = append(notify.Notify, sc.Notifications...)
-	notify.Notify = append(notify.Notify, notifies...)
-	notify.GasConsumed = costGasLimit
+	notify.GasConsumed = tx.GasLimit - sc.Gas
```

Comment:
the gas calculation is greatly simplified. may lead to DOS attack

Reply:
sc.Gas is always less or equal with tx.GasLimit, right?

Reply:
what if tx.GasLimit == MAXINT ?

Reply:
There's no gas charge in DNA.  
There should be one gas-limiter in tx validation.

Reply:
ok, then seems reasonable.